### ISSUE TITLE
build fails with gcc 4.8 on CentOS 7

### ISSUE THREAD

**Issue by dschwoerer** [boutproject/BOUT-dev#1863]
The build fails on CentOS 7:
```
../../../../../include/fieldperp.hxx: At global scope:
../../../../../include/fieldperp.hxx:69:3: warning: unused parameter 'rhs' [-Wunused-parameter]
In file included from ../../../../../include/options.hxx:50:0,
                 from ../../../../../include/bout/deriv_store.hxx:42,
                 from ../../../../../include/bout/mesh.hxx:49,
                 from naulin_laplace.cxx:140:
../../../../../include/field3d.hxx: In function 'FieldPerp pow(const Field3D&, const FieldPerp&, REGION)':
../../../../../include/field3d.hxx:586:37: note: synthesized method 'FieldPerp::FieldPerp(FieldPerp&&)' first required here 
   return pow(lhs, rhs, toString(rgn));
                                     ^
In file included from ../../../../../include/difops.hxx:45:0,
                 from naulin_laplace.cxx:144:
../../../../../include/bout/solver.hxx: At global scope:
../../../../../include/bout/solver.hxx:345:39: error: invalid use of incomplete type 'class std::unique_ptr<T>'
     std::unique_ptr<T> MMS_err{nullptr}; /// Error for MMS
                                       ^
In file included from /usr/include/c++/4.8.2/memory:81:0,
                 from ../../../../../include/bout/array.hxx:31,
                 from ../../../../../include/utils.hxx:36,
                 from ../../../../../include/options.hxx:45,
                 from ../../../../../include/bout/deriv_store.hxx:42,
                 from ../../../../../include/bout/mesh.hxx:49,
                 from naulin_laplace.cxx:140:
/usr/include/c++/4.8.2/bits/unique_ptr.h:109:11: error: declaration of 'class std::unique_ptr<T>'
     class unique_ptr
           ^
make[5]: *** [naulin_laplace.o] Error 1
make[4]: *** [naulin] Error 2
make[4]: *** Waiting for unfinished jobs....
make[3]: *** [impls] Error 2
make[2]: *** [laplace] Error 2
make[1]: *** [invert] Error 2
make: *** [src] Error 2
```

I suspect that gcc 4.8 is to blame.
Do we intend to support 4.8?

How to reproduce:
```
docker run --rm -it centos:7 /bin/bash

yum install git gcc-c++ mpich-devel epel-release
yum install netcdf-cxx-devel netcdf-devel make fftw-devel
yum install $(EDITOR)
useradd build
su build

cd && git clone https://github.com/boutproject/BOUT-dev.git
cd ~/BOUT-dev/ && module load mpi && ./configure && make -j 4
```

**Comment by ZedThree** [comment]
Which branch is this? `next` needs C++14, which needs gcc 5, so not going forwards, no. This looks like it's in `master` though.

It looks like a bug in 4.8 that was fixed in 4.9: https://godbolt.org/z/jjH66j

Personally, I'm fine with not supporting gcc 4.8. Centos 7 users should be able to install EPEL to get more recent compilers. However, if we do need to support gcc 4.8, a workaround is to move the initialisation of `MMS_err` to a constructor:

```cpp
  struct VarStr {
    VarStr() : MMS_err{nullptr} {} // FIXME: remove in v5.0
    bool constraint{false};              /// Does F_var represent a constraint?
    T* var{nullptr};                     /// The evolving variable
    T* F_var{nullptr};                   /// The time derivative or constraint on var
    std::unique_ptr<T> MMS_err;          /// Error for MMS
    ...
```

Although we would have to remember to remove it when merging `master` into `next`.

**Comment by dschwoerer** [comment]
This was the 4.3.0 release with some patches (mostly non-x86_64 related patched)

There is now newer gcc in epel, because policy forbids to override things that are in centos/rhel.

It is fine with me to not support EPEL 7 - but maybe we should mention this in the release notes, that 4.3.x is gcc 4.9+ only ...
